applications/FluidDynamicsApplication/custom_processes/compute_one_side_nodal_pressure_gradient_process.cpp

```diff
@@ -0,0 +1,149 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Mohammad R. Hashemi
+//
+
+// Project includes
+#include "includes/define.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+#include "compute_one_side_nodal_pressure_gradient_process.h"
+
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+
+/// constructor
+ComputeOneSideNodalPressureGradientProcess::ComputeOneSideNodalPressureGradientProcess(
+        ModelPart& rModelPart)
+    : Process(), mrModelPart(rModelPart) {}
+
+/// Constructor with Kratos parameters.
+ComputeOneSideNodalPressureGradientProcess::ComputeOneSideNodalPressureGradientProcess(
+    ModelPart& rModelPart,
+    Parameters Parameters)
+    : ComputeOneSideNodalPressureGradientProcess(
+        rModelPart
+    ){}
+
+/// Constructor with Kratos model
+ComputeOneSideNodalPressureGradientProcess::ComputeOneSideNodalPressureGradientProcess(
+    Model& rModel,
+    Parameters Parameters)
+    : ComputeOneSideNodalPressureGradientProcess(
+        rModel.GetModelPart(Parameters["model_part_name"].GetString())
+    ){}
+
+void ComputeOneSideNodalPressureGradientProcess::Execute(){
+
+    KRATOS_TRY;
+
+    // Set to zero
+    block_for_each(mrModelPart.Nodes(), [&](Node<3>& rNode){
+        rNode.SetValue(NODAL_AREA, 0.0);
+        rNode.FastGetSolutionStepValue(PRESSURE_GRADIENT).clear();
+    });
```

Comment:
I would use VariableUtils `SetHistoricalVariableToZero` and `SetNonHistoricalVariableToZero`

Reply:
but then you have to loop the nodes twice!
I would say this approach is better